lib/omise/OmiseCapabilities.php

```diff
@@ -0,0 +1,192 @@
+    }
+
+    /**
+     * (non-PHPdoc)
+     *
+     * @see OmiseApiResource::g_reload()
+     */
+    public function reload()
+    {
+        parent::g_reload(self::getUrl());
+    }
+
+    /**
+     * Retrieves array of payment backends. Optionally pass in as many filter functions as you want
+     * (muliple arguments, or a single array)
+     *
+     * @param [func1,fun2,...] OR func1, func2,...
+     *
+     * @return array
+     */
+    public function getBackends()
+    {
+        $backends = array_map(
+            function ($backend) {
+                $id = array_keys($backend)[0];
+                $backend[$id]['_id'] = $id;
+                return (object)reset($backend);
+            },
+            $this['payment_backends']
+        );
+
+        // check for filters
+        if ($filters = func_get_args()) {
+            $filter   = self::combineFilters(self::argsToVariadic($filters));
+            $backends = array_filter($backends, $filter);
+            sort($backends);
+        }
+
+        return $backends;
+    }
+
+    /**
+     * Makes a filter function to check supported currency for backend.
+     *
+     * @param  string $currency
+     *
+     * @return function
+     */
+    public function makeBackendFilterCurrency($currency)
+    {
+        return function ($backend) use ($currency) {
+            return in_array(strtoupper($currency), $backend->currencies);
+        };
+    }
+
+    /**
+     * Makes a filter function to check type of backend.
+     *
+     * @param  string $type
+     *
+     * @return function
+     */
+    public function makeBackendFilterType($type)
+    {
+        return function ($backend) use ($type) {
+            return $backend->type == $type;
+        };
+    }
+
+    /**
+     * Makes a filter function to check if backends can handle given amount.
+     *
+     * @param  int $amount
+     *
+     * @return function
+     */
+    public function makeBackendFilterChargeAmount($amount)
+    {
+        $defMin = $this['limits']['charge_amount']['min'];
+        $defMax = $this['limits']['charge_amount']['max'];
+        return function ($backend) use ($amount, $defMin, $defMax) {
+            // temporary hack for now to correct min value for instalments to 500000
+            if ($backend->type == 'installment') {
+                $min = 500000;
+            } else {
+                $min = empty($backend->amount['min']) ? $defMin : $backend->amount['min'];
+            }
+            $max = empty($backend->amount['max']) ? $defMax : $backend->amount['max'];
+            return $amount >= $min && $amount <= $max;
+        };
+    }
+
+    /**
+     * Sets up 'shortcuts' to filters so they may be used thus:
+     *    $capabilities->backendFilter['currency']('THB')
+     * As well as the original:
+     *    $capabilities->makeBackendFilterCurrency('THB')
+     */
+    protected function setupFilterShortcuts()
```

Comment:
Better with _ prefix. Also, why move this function down here, better to leave it near where it was called (in the constructor)? Makes editing the file easier - less jumping around

Reply:
Just moved those protected private stuff down to bottom and moved public methods up above.

Reply:
Disagree with this too. Better to group related methods together